htmlproofer/plugin.py

```diff
@@ -226,9 +229,10 @@ def find_source_file(url: str, src_path: str, files: Dict[str, File]) -> Optiona
             src_dir = urllib.parse.quote(str(pathlib.Path(src_path).parent), safe='/\\')
             search_path = os.path.normpath(str(pathlib.Path(src_dir) / pathlib.Path(url)))
 
-        try:
-            return files[search_path]
-        except KeyError:
+        for file in files:
```

Comment:
Can this impact performance significantly?

Reply:
It changes the O for lookup to be linear so for huge sites, it could have a large impact.

Unfortunately the faster lookup that existed previously is also a bug.

The performance could be addressed by creating a Dictionary at the time that lookup is needed after all the File objects have been fully configured. This would have better performance characteristics but is also more obscure and probably a bit harder to maintain.

On almost all sites, the impact of this check should not be particularly noticeable to a "reasonable human".

Reply:
Note, I'm not sure where the proper place to build that Dictionary to speed the lookups in `find_source_file` would be. 

My python is not the greatest. Ideally, if you want to do that change, I'd prefer to hand it off to you.

Reply:
With the change I had to make to get this to work for the Windows tests, the cost has increased some as `os.path.normpath` needs to be called on `file.url` for the comparison to work, so that is an additional bit of cost.

I ran again locally with the source for ponylang.io (https://github.com/ponylang/ponylang-website) and if there is a difference, it isn't noticeable to me.

Reply:
If I understand the lifecycle correctly then, self.files could be turned into an optimized dictionary in `on_post_page` anytime before the `for a in soup.find_all('a', href=True):` loop.

If that is correct, I do a dictionary based structure that mimics the old data structure but has the correct information.

Can you confirm that my understanding of the lifecycle is correct?

Reply:
Correct!

Reply:
I'll make that change in the not so distant future. Before the end of the weekend at the latest.